src/plugins/layout/stateprovider.ts

```diff
@@ -187,13 +247,16 @@ export function shouldIgnoreNode(node: Node, shadowDom: ShadowDom): boolean {
   // Ignore subtrees of ignored nodes (e.g. text with a <script> parent)
   if (!ignore) {
     let parentIndex = getNodeIndex(node.parentNode);
-    if (parentIndex !== null) {
-      let parentShadowNode = shadowDom.getShadowNode(parentIndex);
-      assert(!!parentShadowNode, "shouldIgnoreNode", "parentShadowNode is missing");
-      if (parentShadowNode && parentShadowNode.ignore && parentShadowNode.node !== document) {
+    if (parentIndex !== null && parentIndex > 0) {
```

Comment:
Yes it can be... That will be a document node, which is what we were checking before as: && parentShadowNode.node !== document. I will add the comment.